core/chains/solana/chain.go

```diff
@@ -0,0 +1,209 @@
+package solana
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/rand"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana"
+	solanaclient "github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/config"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/db"
+	"github.com/smartcontractkit/sqlx"
+
+	"github.com/smartcontractkit/chainlink/core/chains/solana/monitor"
+	"github.com/smartcontractkit/chainlink/core/chains/solana/soltxm"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services"
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/core/services/pg"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+// DefaultRequestTimeout is the default Solana client timeout.
+const DefaultRequestTimeout = 30 * time.Second
+
+//go:generate mockery --name TxManager --srcpkg github.com/smartcontractkit/chainlink-solana/pkg/solana --output ./mocks/ --case=underscore
+//go:generate mockery --name Reader --srcpkg github.com/smartcontractkit/chainlink-solana/pkg/solana/client --output ./mocks/ --case=underscore
+//go:generate mockery --name Chain --srcpkg github.com/smartcontractkit/chainlink-solana/pkg/solana --output ./mocks/ --case=underscore
+var _ solana.Chain = (*chain)(nil)
+
+type chain struct {
+	utils.StartStopOnce
+	id             string
+	cfg            config.Config
+	txm            *soltxm.Txm
+	balanceMonitor services.ServiceCtx
+	orm            db.ORM
+	lggr           logger.Logger
+
+	// tracking node chain id for verification
+	clientChainID map[string]string // map URL -> chainId [mainnet/testnet/devnet/localnet]
+	chainIDLock   *sync.RWMutex
```

Comment:
If you embed the struct, then you can drop all of the `chainIDLock: &sync.RWMutex{}`s.
```suggestion
	chainIDLock   sync.RWMutex
```

Reply:
addressed: https://github.com/smartcontractkit/chainlink/pull/6231/commits/921c8cd0c7b073bdd56fb44f2c0c9443f0498be7